Obtained from:
	https://gitlab.freedesktop.org/libfprint/fprintd/-/commit/0c6bab8640b3586ac8e1b8c35d87aa49b79719d0.patch

From 0c6bab8640b3586ac8e1b8c35d87aa49b79719d0 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 23 Jan 2020 17:37:33 +0100
Subject: [PATCH] main: Fix redeclaration linking error

Fix linking error as the "store" global variable gets redeclared in
each C file that includes the header. Move the actual declaration to
main.c.

Fixes:
 /usr/bin/ld: ./.libs/libfprintd.a(device.o):/builds/libfprint/fprintd/src/storage.h:51: multiple definition of `store'; main.o:/builds/libfprint/fprintd/src/storage.h:51: first defined here
---
 src/main.c    | 2 ++
 src/storage.h | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index 8f7e19c..ce1a47d 100644
--- src/main.c
+++ src/main.c
@@ -33,6 +33,8 @@
 #include "storage.h"
 #include "file_storage.h"
 
+fp_storage store;
+
 extern DBusGConnection *fprintd_dbus_conn;
 static gboolean no_timeout = FALSE;
 static gboolean g_fatal_warnings = FALSE;
diff --git a/src/storage.h b/src/storage.h
index d908c59..28ed2b6 100644
--- src/storage.h
+++ src/storage.h
@@ -46,4 +46,4 @@ struct storage {
 typedef struct storage fp_storage;
 
 /* The currently setup store */
-fp_storage store;
+extern fp_storage store;
-- 
GitLab


